panic in windows

## Describe the bug

when apply in windows , it is panic.

## To reproduce

```console
$ chezmoi apply -v
```

## Output of command with the `--verbose` flag

```console
PS C:\WINDOWS\system32> chezmoi apply -v
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x30 pc=0x994490]

goroutine 1 [running]:
github.com/twpayne/chezmoi/v2/pkg/chezmoi.(*EntryTypeSet).ContainsSourceStateEntry(0xc000791ea8, {0x0, 0x0})
        /home/runner/work/chezmoi/chezmoi/pkg/chezmoi/entrytypeset.go:124 +0x30
github.com/twpayne/chezmoi/v2/pkg/chezmoi.(*EntryTypeFilter).IncludeSourceStateEntry(0xc0004543d0, {0x0, 0x0})
        /home/runner/work/chezmoi/chezmoi/pkg/chezmoi/entrytypefilter.go:34 +0x2e
github.com/twpayne/chezmoi/v2/pkg/chezmoi.(*SourceState).Apply(0xc00022b760, {0x16c8d88, 0xc00050a6c0}, {0x16c8d88, 0xc00050a6c0}, {0x16c6658, 0xc00077d950}, {{0xc0004be1e0?, 0x5cac8f?}}, {{0x0, ...}}, ...)
        /home/runner/work/chezmoi/chezmoi/pkg/chezmoi/sourcestate.go:580 +0xc5
github.com/twpayne/chezmoi/v2/pkg/cmd.(*Config).applyArgs(0xc0007a4000, {0x16c2470, 0xc0000380a8}, {0x16c8d88, 0xc00050a6c0}, {{0xc0004be1e0?, 0x0?}}, {0xc000455d20, 0x0, 0x1}, ...)
        /home/runner/work/chezmoi/chezmoi/pkg/cmd/config.go:561 +0x693
github.com/twpayne/chezmoi/v2/pkg/cmd.(*Config).runApplyCmd(0x0?, 0x0?, {0xc000455d20?, 0x0?, 0x0?})
        /home/runner/work/chezmoi/chezmoi/pkg/cmd/applycmd.go:42 +0xf6
github.com/spf13/cobra.(*Command).execute(0xc0002c9200, {0xc000455d00, 0x1, 0x1})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:916 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0xc0002c8900)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
github.com/twpayne/chezmoi/v2/pkg/cmd.(*Config).execute(0xc00071fdf8?, {0xc00006e050, 0x2, 0x3})
        /home/runner/work/chezmoi/chezmoi/pkg/cmd/config.go:1096 +0x78
github.com/twpayne/chezmoi/v2/pkg/cmd.runMain({{0x16b28c4, 0x6}, {0x16bfd00, 0x28}, {0x16b6480, 0x14}, {0x16b3430, 0xa}}, {0xc00006e050, 0x2, ...})
        /home/runner/work/chezmoi/chezmoi/pkg/cmd/cmd.go:285 +0x22a
github.com/twpayne/chezmoi/v2/pkg/cmd.Main({{0x16b28c4, 0x6}, {0x16bfd00, 0x28}, {0x16b6480, 0x14}, {0x16b3430, 0xa}}, {0xc00006e050, 0x2, ...})
        /home/runner/work/chezmoi/chezmoi/pkg/cmd/cmd.go:108 +0xb1
main.main()
        /home/runner/work/chezmoi/chezmoi/main.go:26 +0x10c
```

## Output of `chezmoi doctor`

<details>

```console
$ chezmoi doctor
```

</details>

--- a/pkg/chezmoi/sourcestate.go
+++ b/pkg/chezmoi/sourcestate.go
@@ -1180,7 +1180,7 @@ func (s *SourceState) addExternal(sourceAbsPath AbsPath) error {
 	s.Lock()
 	defer s.Unlock()
 	for path, external := range externals {
-		if filepath.IsAbs(path) {
+		if strings.HasPrefix(path, "/") || filepath.IsAbs(path) {
 			return fmt.Errorf("%s: %s: path is not relative", sourceAbsPath, path)
 		}
 		targetRelPath := parentTargetSourceRelPath.JoinString(path)

--- a/pkg/cmd/testdata/scripts/issue2302.txtar
+++ b/pkg/cmd/testdata/scripts/issue2302.txtar
@@ -1,5 +1,3 @@
-[windows] skip 'UNIX only'
-
 # test that chezmoi does not panic if an external uses an absolute path
 ! exec chezmoi diff
 stderr 'path is not relative'

